ui/app/components/oidc/key-form.js

```diff
@@ -30,6 +30,11 @@ export default class OidcKeyForm extends Component {
       ? 'allow_all'
       : 'limited';
 
+  get filterDropdownOptions() {
+    // query object sent to search-select so only clients that reference this key appear in dropdown
+    return { paramKey: 'key', filterFor: [this.args.model.name] };
```

Comment:
the `named-path` adapter can accept `paramKey` and `filterFor` if the response contains `key_info`